### PR TITLE
setup e2e testing using playwright

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

Adds a new `e2e` workspace, where [Playwright](https://playwright.dev/) e2e testing has been set up in a Remix app.

It actually turned out to be quite straightforward specifically because of [remix's file-based routing](https://remix.run/docs/en/main/discussion/routes). To add a new test, we only need to create a new route and add a `spec.ts` file to visit that route! Each route/test can be completely independent of one another.

This PR is just opening the path to adding tests in the future. No real tests have been added so far.

## Testing

Tested a hello world scenario locally (`pnpm run test --filter=e2e`) and in CI.

## Docs

Added basic instructions in `CONTRIBUTING.md`.

### CODE REVIEW

File: testing/e2e/tests-examples/demo-todo-app.spec.ts
Comment (mayank99) [inline testing/e2e/tests-examples/demo-todo-app.spec.ts:1]: I've deleted this example test that Playwright created upon initialization, but it's worth skimming through, just to understand how to do some pretty neat things with playwright.

File: .vscode/extensions.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "streetsidesoftware.code-spell-checker",
     "esbenp.prettier-vscode",
     "dbaeumer.vscode-eslint",
-    "astro-build.astro-vscode"
+    "astro-build.astro-vscode",
+    "ms-playwright.playwright"
```
Comment (mayank99) [inline .vscode/extensions.json:8]: The extension is pretty neat

https://github.com/iTwin/iTwinUI/assets/9084735/7be6e5e4-5c81-40a7-a188-37c86b89316a

Comment (Ben-Pusey-Bentley) [reply .vscode/extensions.json:8]: Wow, that seems incredibly useful for debugging tests.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -384,6 +384,15 @@ In Cypress, if the component violates a Axe rule, its rule ID and the number of
 
 For more information on the Axe rule IDs and their meanings, visit [Deque University's list of Axe rules.](https://dequeuniversity.com/rules/axe/4.4/)
 
+### E2E testing
+
+The `testing/e2e` workspace facilitates testing of complex scenarios in a real browser. This is achieved by running [Playwright](https://playwright.dev/) tests against a [Remix](https://remix.run/) app.
+
+- **To run tests**, use this command: `pnpm run test --filter=e2e`.
```
[the remark this answers is not in the document]
Comment (mayank99) [reply CONTRIBUTING.md:391]: You mean running `pnpm install` was not enough?

[the remark this answers is not in the document]
Comment (rohan-kadkol) [reply CONTRIBUTING.md:391]: Yes, `pnpm install` was not sufficient for me.

[the remark this answers is not in the document]
Comment (mayank99) [reply CONTRIBUTING.md:391]: Should work fine after latest commit.

@Ben-Pusey-Bentley Can you also test?

[the remark this answers is not in the document]
Comment (Ben-Pusey-Bentley) [reply CONTRIBUTING.md:391]: Can confirm that this is working for me now.

**Review by Ben-Pusey-Bentley — APPROVED** [review]
LGTM.

**Pull request merged**